index.bs

```diff
@@ -3448,15 +3423,19 @@ To <dfn>trigger aggregatable attribution</dfn> given an [=attribution trigger=]
 
 <h3 algorithm id="triggering-attribution">Triggering attribution</h3>
 
-To <dfn>obtain and deliver a debug report on trigger registration</dfn> given a [=trigger debug data type=] |dataType|,
-an [=attribution trigger=] |trigger| and an optional [=attribution source=]
+To <dfn>obtain and deliver a debug report on trigger registration</dfn>
+given a [=set=] of [=trigger debug data=] |dataSet|, an [=attribution trigger=] |trigger|,
```

Comment:
What do you think about making this a list instead of a set? I ask because it's not clear to me how the equality operation is supposed to work for `trigger debug data`, particularly with respect to its `report` field.

Reply:
The comparison is field-by-field, including the report field, which is mostly null except for two event errors. So basically it’s just comparing types.

Reply:
Is there actually a benefit to using a set though? As written, it seems like implementations will need to compare every single field in the report in this algorithm.

Reply:
As event-level and aggregate may have the same debug type, so using a set will deduplicate them and only results in one payload. Alternately, we can pass as a list and deduplicate based on types in the report generation algorithm. Wdyt?

Reply:
Could you clarify when they can have the same debug type? I thought that the common debug types were only generated before calling the aggregatable-specific and event-level-specific algorithms.

Reply:
I think it would be clearer if we refactored [that algorithm](https://pr-preview.s3.amazonaws.com/linnan-github/conversion-measurement-api/pull/1285.html#check-if-attribution-should-be-blocked-by-rate-limits) to perform [should processing be blocked by reporting-origin limit](https://pr-preview.s3.amazonaws.com/linnan-github/conversion-measurement-api/pull/1285.html#should-block-processing-for-reporting-origin-limit) only once per attribution, instead of once for event-level and once for aggregatable, since IIUC the result should be the same.

If that's not possible because the checks are done in a particular order for privacy reasons, then I would expect that algorithm to have a note making that clear.

Reply:
Added an issue to considering refactoring to make this PR no behavioral change.

Reply:
Could you also add an issue for changing the type from `set` to `list`?

Reply:
Added an issue, thanks.